### DISCUSSION TITLE
use session

### DISCUSSION THREAD

**Discussion by Ali-Nazarii** [jsocol/django-ratelimit#237] [Q&A]
Hello, how can I use the session as a key when the user has not logged in?

**Comment by jsocol** [comment]
Hey @Ali-Nazarii—this isn't directly supported, so the best option is probably going to be to use a [callable `key`](https://django-ratelimit.readthedocs.io/en/stable/keys.html#callable-values):

```py
def session_key(group, request):
    key = request.session.get('some_key', None)
    if key is None:
        key = generate_key_somehow()
        request.session['some_key'] = key
    return key

@ratelimit(key=session_key)
def my_view(req):
   pass
```

You could use the session ID directly, but be aware that not all backends have a session ID and that it's not necessarily stable (actions like logging in or out might rotate the session key).
